src/components/data-exploration/generic-gene-table/GeneTable.jsx

```diff
@@ -7,47 +7,70 @@ import _ from 'lodash';
 import PropTypes from 'prop-types';
 import { changeGeneSelection } from 'redux/actions/genes';
 import GeneSelectionStatus from 'redux/actions/genes/geneSelectionStatus';
-import { geneTableUpdateReason } from 'utils/geneTable/geneTableUpdateReason';
 import FocusButton from 'components/FocusButton';
 import PlatformError from 'components/PlatformError';
-import useLazyEffect from 'utils/customHooks/useLazyEffect';
 import GeneSelectionMenu from 'components/data-exploration/generic-gene-table/GeneSelectionMenu';
 import FilterGenes from 'components/data-exploration/generic-gene-table/FilterGenes';
 import Loader from 'components/Loader';
 
+const valueComparator = (key) => (a, b) => {
```

Comment:
Can you tell me why you need this function?

Reply:
This is used for sorting the columns. The way react tables sort the values is that you pass in a comparator function to it and upon click items are compared against each other in order to get sorted. 

Here is how the two tables (Gene list and Differential expression) are created with different columns:
```
const newColumns = propColumns.map((column) => {
  const modifiedColumn = { ...column, dataIndex: column.key };

  if (column.sorter) {
    modifiedColumn.sorter = valueComparator(column.key);
    modifiedColumn.sortOrder = getSortOrder(column.key);
  }

  return modifiedColumn;
});
```

As their columns have different types I created a generic comparator which handles both strings and numbers. As the column type is not known at the time of creation this solution made the most sense to me.